### PR TITLE
Add Unclutter 2.1.22d

### PR DESCRIPTION
**Pull request by NanoSector**
This cask has been submitted a few times before but rejected because it was unclear if the app could be purchased from within the app and not requiring a download: #46823 #37546

I have bought the app and it is possible to activate it with an activation link from the e-mail you receive afterwards. The app also mentions activation using a license file. It does not require a re-download, clicking the link brings you back to Unclutter which then activates and removes the trial restrictions. Therefore I don't think it differs from apps like Little Snitch or Alfred which are included in the repo.

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

File: Casks/unclutter.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cask 'unclutter' do
+  version :latest
+  sha256 :no_check
```
Comment (ran-dall) [inline Casks/unclutter.rb:3]: Needs a `version` preferably.

File: Casks/unclutter.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+  url 'https://unclutterapp.com/files/Unclutter.zip'
+  name 'Unclutter'
+  homepage 'https://unclutterapp.com/'
+
+  app 'Unclutter.app'
+end
```
Comment (ran-dall) [inline Casks/unclutter.rb:10]: Missing a `zap` stanza.

**Review by ran-dall — CHANGES_REQUESTED** [review]

File: Casks/unclutter.rb (hunk 1 above)
Comment (Yoshi2889) [reply Casks/unclutter.rb:3]: The site only gives out an Unclutter.zip without version info. I could version and hash this but whenever they push out an update it’ll break this cask, I’m not sure that’s preferable. Maybe there’s a different endpoint it uses for updates, I’ll have to check.

File: Casks/unclutter.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+cask 'unclutter' do
+  version :latest
+  sha256 :no_check
+
+  url 'https://unclutterapp.com/files/Unclutter.zip'
```
Comment (vitorgalvao) [inline Casks/unclutter.rb:5]: Please add an [`appcast`](https://github.com/homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md): `https://unclutterapp.com/files/Unclutter.zip`

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/unclutter.rb (hunk 1 above)
Comment (vitorgalvao) [reply Casks/unclutter.rb:3]: Usually you’d be right. But because we can have an `appcast`, it can be versioned.

**Comment by vitorgalvao** [comment]
> but rejected because it was unclear if the app could be purchased from within the app

No, it was rejected because *it was clear it couldn’t* be purchased from outside the MAS. It’s irrelevant if it can be done from inside the app.

Right now there’s a clear direct payment option, so it can be accepted.

**Comment by ran-dall** [comment]
@Yoshi2889 All you're missing now is a `zap` stanza.

**Comment by Yoshi2889** [comment]
I’ll take a look at the zap stanza later tonight, thank you for the feedback and changes!

**Comment by Yoshi2889** [comment]
I've added a zap stanza as requested; I wasn't sure if the `/var/folders` entries should also be included, I remember reading that macOS cleans those up automatically on reboot.

I mainly used AppCleaner to scan for these as it seems to usually be fairly accurate, this is what it came up with:

image

Please let me know if I missed anything.

**Pull request merged**